tensorflow_addons/seq2seq/decoder.py

```diff
@@ -253,16 +257,16 @@ def tracks_own_finished(self):
 
 
 def dynamic_decode(
-    decoder,
-    output_time_major=False,
-    impute_finished=False,
-    maximum_iterations=None,
-    parallel_iterations=32,
-    swap_memory=False,
-    training=None,
-    scope=None,
+    decoder: BaseDecoder,
```

Comment:
As discussed, this should be a union:

```suggestion
    decoder: Union[Decoder, BaseDecoder],
```

Reply:
Oh, yes, updated. Thanks @guillaumekln.